features/integration/abi.feature

```diff
@@ -0,0 +1,80 @@
+    And I build a method with signature "<method-signature>".
+    And I prepare the method arguments with the "<app-args>".
+    And I add a method call with the signing account, the current application, suggested params, operation "call", current transaction signer, current method arguments.
+    # Build the group in the composer
+    And I build the transaction group with the composer. If there is an error it is "".
+    Then The composer should have a status of "BUILT".
+    And I gather signatures with the composer.
+    Then The composer should have a status of "SIGNED".
+    And I submit the current transaction group with the composer.
+    Then The composer should have a status of "SUBMITTED".
+    # Execute the group and check that the log contains the correct return value
+    And I execute the current transaction group with the composer.
+    Then The composer should have a status of "COMMITTED".
+    And The app should have returned "<returns>" in the log.
+
+    Examples:
+      | method-signature         | app-args                                                  | returns |
+      | add(uint64,uint64)uint64 | [int:1,int:1]                                             | 151f7c750000000000000002 |
+      | empty()void              | []                                                        | 151f7c75 |
+
+  Scenario: AtomicTransactionComposer with transaction arguments test
+    Given a new AtomicTransactionComposer
+    And I fund the signing account with 100000000 microalgos.
+    # Application create with ARC-0004 compliant app should succeed with expected message. 
+    And I build an application transaction with the signing account, the current application, suggested params, operation "create", approval-program "programs/abi_method_call.teal.tok", clear-program "programs/one.teal.tok", global-bytes 0, global-ints 0, local-bytes 0, local-ints 0, app-args "", foreign-apps "", foreign-assets "", app-accounts "", extra-pages 0
+    And I sign and submit the transaction, saving the txid. If there is an error it is "".
+    And I wait for the transaction to be confirmed.
+    And I remember the new application ID.
+    # Create a pay transaction, create a TransactionSigner, and add it to the composer
+    When I make a transaction signer for the signing account.
+    And I remember the transaction signer.
+    And I build a payment transaction with sender "BH55E5RMBD4GYWXGX5W5PJ5JAHPGM5OXKDQH5DC4O2MGI7NW4H6VOE4CP4", receiver "BH55E5RMBD4GYWXGX5W5PJ5JAHPGM5OXKDQH5DC4O2MGI7NW4H6VOE4CP4", amount 100001, close remainder to ""
+    And I create a transaction with signer with the current transaction.
+    And I remember the transaction with signer.
+    # Create a payment method call with an address argument, and add it to the composer
+    And I build a method with signature "<method-signature>".
+    And I prepare the method arguments with the current transaction with signer and the "<app-args>".
```

Comment:
I like the steps for method app args, but I'm not a fan of the argument format. Prefixes like `int:` and `hex:` work for encoding values to bytes for normal app calls, but I don't think they work as well for ABI types. We could implement general JSON parsing and conversion like goal does in https://github.com/algorand/go-algorand/pull/3088, that's a lot of work just for testing.

What if instead the argument format was a comma-separated list of the encoded ABI values in base64? For example, the args 2 and 3 for the method `add(uint64,uint64)uint64` could be given as `AAAAAAAAAAI=,AAAAAAAAAAM=`. It's not the nicest to look at but it would be pretty simple to decode and process in this step, even for complicated ABI values.

Reply:
Sounds good to me 👍

Reply:
Actually, if I'm think about this correctly, method arguments are in their native types in the respective SDKs. 

```
type ABIValue = boolean | number | bigint | string | Uint8Array | ABIValue[];
type MethodArgument = ABIValue | TransactionWithSigner;
```

So when `addMethodCall()` for `add` is called on the composer, it's passed in `[1, 1]`, where `1` is an `int` or `number`, etc. I'm inclined to go back to the prefix format so it can just cast the value based on that, but interested to hear your thoughts.

Reply:
edit: I think the ABI encoded values makes sense